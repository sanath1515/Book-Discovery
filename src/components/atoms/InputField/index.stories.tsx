import React,{ComponentProps} from "react";
import {Story} from "@storybook/react";
import {action} from "@storybook/addon-actions";
import InputField,{TextFieldProps} from ".";

const InputStory={
    title:"Atoms/Input",
    component:InputField,
};


const Template: Story<ComponentProps<typeof InputField>>=(args:TextFieldProps)=>
    <InputField {...args} />;

export const Input = Template.bind({});
Input.args={
    placeholder:'InputField',
    onChange:action('Input changed'),
    disableUnderline:true,
    autoFocus:true,
    size:'small',
};

export const UnderlinedTextField= Template.bind({});
UnderlinedTextField.args={
    placeholder:'InputField',
    onChange:action('Input changed'),
    disableUnderline:false,
    autoFocus:false,
    size:'medium',
};

export const NumberTextField= Template.bind({});
NumberTextField.args={
    placeholder:'InputField',
    onChange:action('Input changed'),
    disableUnderline:false,
    type:'number',
};
export default InputStory;