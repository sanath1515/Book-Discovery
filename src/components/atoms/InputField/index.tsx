import TextField from '@material-ui/core/TextField';
import React from "react";

export type TextFieldProps={
  placeholder?:string;
  value?:unknown;
  type?:string;
  size?:'small'|'medium';
  disableUnderline?:boolean;
  autoFocus?:boolean;
  onChange?:React.ChangeEventHandler<HTMLTextAreaElement | HTMLInputElement>;
  className?:string;
}

const InputField:React.FC<TextFieldProps>= ({placeholder,size,autoFocus,value,type,onChange,disableUnderline,className}) => {
  return (
      <TextField value={value} size={size}  placeholder={placeholder} autoFocus={autoFocus} onChange={onChange} className={className} type={type} InputProps={{ disableUnderline: disableUnderline,type:'search' }} fullWidth />
  );

}

export default InputField;