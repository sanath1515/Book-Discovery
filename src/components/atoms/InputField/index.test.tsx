import {render,fireEvent} from "@testing-library/react";
import InputField from ".";

describe("Input Field test",()=>{

    const handleChange=jest.fn();

    it("should render input correctly",()=>{
        const {container}=render(<InputField   value="text" onChange={handleChange}/>)
        expect(container.querySelector("input")).toBeTruthy();
    });

    it("should match the snapshot",()=>{
        const {container}=render(<InputField    value="text" onChange={handleChange}/>)
        expect(container.querySelector("input")).toMatchSnapshot();
    });

    it("should call handle change when input value is changed",()=>{
        const {container}=render(<InputField  value={""} onChange={handleChange}/>)
        const input=container.querySelector("input");
        {input&& fireEvent.change(input, {target:{value:'value'} })}
        expect(handleChange).toHaveBeenCalledTimes(1);
    });

    
})