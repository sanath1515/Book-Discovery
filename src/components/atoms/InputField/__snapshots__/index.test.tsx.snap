// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Field test should match the snapshot 1`] = `
<input
  aria-invalid="false"
  class="MuiInputBase-input MuiInput-input MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch"
  type="search"
  value="text"
/>
`;
